django/db/models/sql/compiler.py

```diff
@@ -358,7 +358,7 @@ def get_order_by(self):
                         field, self.query.get_meta(), default_order=asc,
                     ))
             else:
-                if col not in self.query.extra_select:
+                if col not in self.query.extra_select and not self.query.combinator:
```

Comment:
This can be simplified to--

```python
if col in self.query.extra_select or self.query.combinator:
```

by changing the order of the two cases.